plugins/cliconf/nxos.py

```diff
@@ -409,14 +409,14 @@ def _get_command_with_output(self, command, output):
                 % (output, ",".join(options_values["output"])),
             )
 
-        if output == "json" and not command.endswith("| json"):
+        if output in ["json", "json-pretty"] and not command.endswith("| json"):
```

Comment:
Do you need to modify the `endswith` check, too? Maybe also the one on line 420?